src/matching/blossomv.jl

```diff
@@ -0,0 +1,68 @@
+"""
+    minimum_weight_perfect_matching{T<:Real}(g, w::Dict{Edge,T})
+    minimum_weight_perfect_matching{T<:Real}(g, w::Dict{Edge,T}, cutoff)
+
+Given a graph `g` and an edgemap `w` containing weights associated to edges,
+returns a matching with the mimimum total weight among the ones containing
+exactly `nv(g)/2` edges.
+
+Edges in `g` not present in `w` will not be considered for the matching.
+
+This function relies on the BlossomV.jl package, a julia wrapper
+around Kolmogorov's BlossomV algorithm.
+
+Eventually a `cutoff` argument can be given, to the reduce computational time
+excluding edges with weights higher than the cutoff.
+
+The returned object is of type `MatchingResult`.
+"""
+function minimum_weight_perfect_matching end
+
+function minimum_weight_perfect_matching{T<:Real}(g::Graph, w::Dict{Edge,T}, cutoff)
+    wnew = Dict{Edge, T}()
+    for (e, c) in w
+        if c <= cutoff
+            wnew[e] = c
+        end
+    end
+    return minimum_weight_perfect_matching(g, wnew)
+end
+
+function minimum_weight_perfect_matching{T<:AbstractFloat}(g::Graph, w::Dict{Edge,T})
+    wnew = Dict{Edge, Int32}()
+    cmax = maximum(values(w))
+    cmin = minimum(values(w))
+    tmax = typemax(Int32)  / 10. # /10 is kinda arbitrary,
+                                # hopefully high enough to not incurr in overflow problems
+    for (e, c) in w
+        wnew[e] = round(Int32, (c-cmin) / (cmax-cmin) * tmax)
+    end
+    match = minimum_weight_perfect_matching(g, wnew)
+    weight = T(0)
+    for i=1:nv(g)
+        j = match.mate[i]
+        if j > i
+            weight += w[Edge(i,j)]
+        end
+    end
+    return MatchingResult(weight, match.mate)
+end
+
+function minimum_weight_perfect_matching{T<:Integer}(g::Graph, w::Dict{Edge,T})
+    m = BlossomV.Matching(nv(g))
+    for (e, c) in w
+        BlossomV.add_edge(m, src(e)-1, dst(e)-1, c)
+    end
```

Comment:
Weird. I need to read up on how that works then. Thanks for testing and ignore my comment above :)